ospf6d/ospf6_area.c

```diff
@@ -929,41 +941,59 @@ DEFUN (show_ipv6_ospf6_area_spf_tree,
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ipv6_ospf6_simulate_spf_tree_root,
-       show_ipv6_ospf6_simulate_spf_tree_root_cmd,
-       "show ipv6 ospf6 simulate spf-tree A.B.C.D area A.B.C.D",
-       SHOW_STR
-       IP6_STR
-       OSPF6_STR
-       "Shortest Path First calculation\n"
-       "Show SPF tree\n"
-       "Specify root's router-id to calculate another router's SPF tree\n"
-       "OSPF6 area parameters\n"
-       OSPF6_AREA_ID_STR)
+DEFUN(show_ipv6_ospf6_area_spf_tree, show_ipv6_ospf6_area_spf_tree_cmd,
+      "show ipv6 ospf6 [vrf <NAME|all>] area A.B.C.D spf tree",
+      SHOW_STR IP6_STR OSPF6_STR VRF_CMD_HELP_STR
+      "All VRFs\n" OSPF6_AREA_STR OSPF6_AREA_ID_STR
+      "Shortest Path First calculation\n"
+      "Show SPF tree\n")
 {
-	int idx_ipv4 = 5;
-	int idx_ipv4_2 = 7;
+	int idx_ipv4 = 4;
 	uint32_t area_id;
+	struct ospf6 *ospf6;
+	struct listnode *node;
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (idx_vrf > 0)
+		idx_ipv4 += 2;
+
+	if (inet_pton(AF_INET, argv[idx_ipv4]->arg, &area_id) != 1) {
+		vty_out(vty, "Malformed Area-ID: %s\n", argv[idx_ipv4]->arg);
+		return CMD_SUCCESS;
+	}
+
+	if (vrf_name) {
+		if (all_vrf) {
+			for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+				show_ospf6_area_spf_tree_common(
+					vty, argv, ospf6, area_id, idx_ipv4);
+			}
+			return CMD_SUCCESS;
+		}
+		ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+		OSPF6_CMD_CHECK_RUNNING(ospf6);
+		show_ospf6_area_spf_tree_common(vty, argv, ospf6, area_id,
```

Comment:
Same comment as before. This call for specific VRFs could also have been accommodated inside the for loop for 'all_vrf'.

Reply:
Resolved.